### ISSUE TITLE
How to set max_lengh of tag field greater than 50 ?

### ISSUE THREAD

**Issue by BitcoinKing** [radiac/django-tagulous#17]
Tag field throws an error if string is long (greater than 50):
DatabaseError: value too long for type character varying(50)

So i cannot save data. How to set max_length of tag field greater than 50 ?

The same problem with simple fields on SO:
http://stackoverflow.com/questions/3803552/django-fixture-fails-stating-databaseerror-value-too-long-for-type-character

**Comment by radiac** [comment]
I wonder if it might be the slug field on the tag model - you're seeing this on save, rather than on migrate?

I'll look into it, but in the meantime can you show us a full traceback for the exception?

**Comment by GraphGrail** [comment]
Yes, on save, not migrate.
So, how to fix slug? Saw other issues but there are no resolve

**Comment by radiac** [comment]
Yes, I feel terrible - the issue was raised ages ago, but my circumstances changed and I never got time to come back and sort it out.

I know what needs doing - the tag model's slugify routine (https://github.com/radiac/django-tagulous/blob/master/tagulous/models/models.py#L434) needs to be aware of the slug field's limit, to make sure it never generates a slug which is longer than that limit. If you'd like to tackle that you're welcome to submit a PR, otherwise I'll try to take a look over the weekend and get another release out.

In the meantime, you could either create a custom tag model where you override the slug field with a longer max length, or we do have an outstanding PR from PamelaM where there is a flag to override the slug length - you could install her branch directly from github if you want (although if you do, keep an eye on the PR for updates on how to upgrade to the next release - the settings probably won't be exactly the same when it gets merged).

**Comment by GraphGrail** [comment]
The code is hard, so i simply need example on how to create custom tag model where you override the slug field with a longer max length. Browsed docs, but couldn't find

**Comment by radiac** [comment]
This untested hack might work:

```
class MyModel(models.Model):
    tags = TagField(...)
MyModel.tags.tag_model._meta.get_field('slug').max_length=255
```

This should change the slug field's max length after it has been defined, but before it gets noticed by migrations. It's still not perfect, but it should buy you some time.

If it does work, keep an eye on this ticket for the proper fix, so you can remove that hack. I started on a proper fix but have run out of time this weekend I'm afraid.

**Comment by radiac** [comment]
Sorry, ignore previous (deleted) comment - it would put your code in a bit of a mess in the longer term. I think the only good option is for me to try to get this fix released later today or tomorrow. Will get back to you.

**Comment by GraphGrail** [comment]
I have done this manually by ALTER TABLE sql

**Comment by radiac** [comment]
That will work, but in the long term it would be worth changing your model to match the database. The develop branch now has the setting `TAGULOUS_SLUG_MAX_LENGTH` - there are a few outstanding issues to resolve before I can publish version 0.12.0, so you can either switch to the develop branch to use it now:

    pip uninstall tagulous
    pip install -e git+https://github.com/radiac/django-tagulous.git@develop#egg=django-tagulous

or you can wait until 0.12.0 is on pypi; I'll update this issue when it is released.

**Comment by radiac** [comment]
For info, the new setting `TAGULOUS_SLUG_MAX_LENGTH` is now available in 0.12 which has been released.
